File: app/src/main/java/com/example/androidscript/Activities/Menu.java
package com.example.androidscript.Activities;

import com.example.androidscript.FloatingWidget.FloatingWidgetService;
import com.example.androidscript.Activities.ArkKnights.ArkKnightsEditor;
import com.example.androidscript.util.AutoClick;
import com.example.androidscript.util.BtnMaker;

import android.content.Intent;
import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

import android.util.DisplayMetrics;
import android.widget.Button;

import com.example.androidscript.R;
import com.example.androidscript.util.FileOperation;

import org.opencv.android.OpenCVLoader;

import com.example.androidscript.util.ScreenShot;

public class Menu extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        FileOperation.fileRootInit(getExternalMediaDirs()[0].getAbsolutePath() + "/");
        DisplayMetrics displayMetrics = new DisplayMetrics();
        getWindowManager().getDefaultDisplay().getRealMetrics(displayMetrics);
        ScreenShot.setUpScreenDimension(displayMetrics.heightPixels, displayMetrics.widthPixels);
        setContentView(R.layout.activity_menu);
        BtnMaker.Jump(R.id.button_to_ArkUI, this, ArkKnightsEditor.class);
        BtnMaker.JumpWithMessage(R.id.button_to_FGO, this, SelectFile.class, "next_destination", "com.example.androidscript.Activities.FGO.FGOEditor");

        Button btn = findViewById(R.id.button_to_basic_landscape);
        btn.setOnClickListener((v -> {
            Intent intent = new Intent(this,SelectFile.class );
            intent.putExtra("next_destination", "com.example.androidscript.Activities.Basic.BasicEditor");
            intent.putExtra("Orientation","Landscape");
            startActivity(intent);
        }));

        btn = findViewById(R.id.button_to_basic_portrait);
        btn.setOnClickListener((v -> {
            Intent intent = new Intent(this,SelectFile.class );
            intent.putExtra("next_destination", "com.example.androidscript.Activities.Basic.BasicEditor");
            intent.putExtra("Orientation","Portrait");
            startActivity(intent);
        }));

        BtnMaker.registerOnClick(R.id.Exit, this, v -> endService(true));
        if (!OpenCVLoader.initDebug()) {
            throw new AssertionError("OpenCV unavailable!");
        }
    }

    @Override
    protected void onResume() {
        super.onResume();
        if (getIntent().getStringExtra("Message") != null && getIntent().getStringExtra("Message").equals("Reset")) {
            endService(false);
        }
    }
    public void endService(boolean stop){
        AutoClick.stop();
        ScreenShot.endProjection();
        stopService(new Intent(this,FloatingWidgetService.class));
        stopService(new Intent(this, ScreenShot.class));
        if(stop){
            this.finish();
        }
    }
}

File: app/src/main/java/com/example/androidscript/activities/Menu.kt
package com.example.androidscript.activities

import android.content.Intent
import android.os.Build
import android.os.Bundle
import android.util.DisplayMetrics
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.example.androidscript.R
import com.example.androidscript.activities.arknights.ArkKnightsEditor
import com.example.androidscript.activities.basic.BasicEditor
import com.example.androidscript.activities.fgo.FGOEditor
import com.example.androidscript.services.ScreenShotService
import com.example.androidscript.util.FileOperation
import org.opencv.android.OpenCVLoader

/** App entry
 *
 * There are 4 options:
 *  Arknights(Landscape)
 *  FGO(Portrait)
 *  Portrait
 *  Landscape
 *
 * Since screen casting will need to decide the shape, we split Basic to two button here.
 *
 * TODO Consider dynamically reset casting orientation as a new feature.
 */
class Menu : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        if (!OpenCVLoader.initDebug()) {
            throw AssertionError("OpenCV unavailable!")
        }

        // TODO stop service
        // Pre-setup
        // TODO make activity to handle it's own root.
        // A known issue is that when we re-enter this activity, onCreate won't be called
        // and root is not initialized.
        FileOperation.root = (externalMediaDirs[0].absolutePath + "/")

        @Suppress("DEPRECATION") // For old api version
        windowManager.run {
            if (Build.VERSION.SDK_INT < Build.VERSION_CODES.R) {
                DisplayMetrics().also {
                    defaultDisplay.getRealMetrics(it)
                }.run {
                    ScreenShotService.setScreenDim(heightPixels, widthPixels)
                }
            } else currentWindowMetrics.bounds.run {
                ScreenShotService.setScreenDim(height(), width())
            }
        }

        // View-setup
        setContentView(R.layout.activity_menu)

        findViewById<View>(R.id.button_to_ArkUI).setOnClickListener {
            ScreenShotService.setShotOrientation(isLandscape = true)
            startActivity(Intent(this, ArkKnightsEditor::class.java))
        }

        // TODO combine SelectFileActivity.kt with this, so that no putExtra is needed.
        findViewById<View>(R.id.button_to_FGO).setOnClickListener {
            ScreenShotService.setShotOrientation(isLandscape = true)
            val intent = Intent(this, SelectFileActivity::class.java)
            intent.putExtra(
                "next_destination",
                FGOEditor::class.qualifiedName
            )
            startActivity(intent)
        }

        findViewById<View>(R.id.button_to_basic_landscape).setOnClickListener {
            ScreenShotService.setShotOrientation(isLandscape = true)
            val intent = Intent(this, SelectFileActivity::class.java)
            intent.putExtra(
                "next_destination",
                BasicEditor::class.qualifiedName
            )
            startActivity(intent)
        }

        findViewById<View>(R.id.button_to_basic_portrait).setOnClickListener {
            ScreenShotService.setShotOrientation(isLandscape = false)
            val intent = Intent(this, SelectFileActivity::class.java)
            intent.putExtra(
                "next_destination",
                BasicEditor::class.qualifiedName
            )
            startActivity(intent)
        }

        // TODO Remove this button.
        findViewById<View>(R.id.Exit).setOnClickListener {
            finishAffinity()
        }
    }
}